solarextrapolation/potential_extrapolator.py

```diff
@@ -0,0 +1,119 @@
+    def __init__(self, map_magnetogram, **kwargs):
+        super(PotentialExtrapolator, self).__init__(map_magnetogram, **kwargs)
+        self.meta['extrapolator_routine'] = 'Potential Field Extrapolator'
+
+    def _extrapolation(self):
+        # Adding in custom parameters to the meta
+        phi = self._extrapolate_phi()
+        Bxyz = self._determine_vec(phi, D = 1, debug = False)
+
+        return Map3D( Bxyz, self.meta )
+
+    # Greens function.
+    def _Gn_5_2_26(self, inR, inRPrime):    
+        floModDr = np.linalg.norm(inR - inRPrime)
+        floOut = 1.0 / ( 2.0 * np.pi * floModDr)
+        return floOut
+    
+    # Greens function.
+    def _Gn_5_2_29(self, i, j, k, i_prime, j_prime, d, d_com):    
+        d_i = i - i_prime
+        d_j = j - j_prime
+        d_k = k - d_com # d / np.sqrt(2.0 * np.pi)
+        floModDr = np.sqrt( d_i * d_i + d_j * d_j + d_k * d_k)
+        
+        #print 'floModDr: ' + str(floModDr)
+        floOut = 1.0 / ( 2.0 * np.pi * floModDr)
+        #print 'floOut: ' + str(floOut)
+        return floOut
+    
+    
+    # A function to extrapolate the magnetic field above the given boundary.
+    def _phi_extrapolation_python(self, boundary, d):    
+        # Volume size.
+        M = boundary.shape[0]
+        N = boundary.shape[1]
+        Z = self.z
+        
+        # Derived parameters
+        d = 1.0
+        d_squ = np.power(d,2)
+        d_com = d / np.sqrt(2.0 * np.pi)
+        
+        # Create the empty numpy volume array.
+        D = np.empty((M, N, Z), dtype=np.float)
+        
+        # Itherate though the 3D space.
+        for i in range(0, M):
+            for j in range(0, N):
+                for k in range(0, Z):
+                    # Variable holding running total for the contributions to point.
+                    point_phi_sum = 0.0
+                    # Iterate through the boundary data.
+                    for i_prime in range(0,M):
+                        for j_prime in range(0,N):                        
+                            # Find the components for this contribution product.
+                            B_n = boundary[i_prime, j_prime]
+                            
+                            G_n = self._Gn_5_2_29(i, j, k, i_prime, j_prime, d, d_com)
+    
+                            # Add the contribution.
+                            point_phi_sum += B_n * G_n * d_squ
+                    # Now add this to the 3D grid.
+                    D[i, j, k] = point_phi_sum
+        return D
+    
+    def _extrapolate_phi(self, debug=False):
+        '''
+        A function to extrapolate the magnetic field above the given boundary.
+        Assumes the input B-field boundary data is near normal (the image must
+        be near the centre of the HMI data).
+        P183 (5.2.28)
+        '''
+        if debug:
+            print 'extrapolatePhi(' + str(self.map_boundary_data.data.shape) + ', ' + str(inZ) + ', ' + str(debug) + ')'
```

Comment:
instead of concatenating strings with `+` it is a  better idea to use string formatting:
i.e.

```python
print "extrapolatePhi({},{},{})".format(self.map_boundary_data.data.shape, inZ, debug)
```